src/display/accessibility.js

```diff
@@ -0,0 +1,262 @@
+/* Copyright 2022 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { binarySearchFirstItem } from "./display_utils.js";
+import { shadow } from "../shared/util.js";
+
+const DEBUG = false;
+
+/**
+ * This class aims to provide some methods:
+ *  - to reorder element in the DOM with respect to the visual order;
+ *  - make a link in using aria-owns between spans in the text layer and
+ *    annotations in the annotation layer. The goal is to help to know
+ *    where the annotations are in the text flow.
+ */
+class AccessibilityManager {
```

Comment:
Should this class perhaps be named `TextAccessibilityManager` instead, and the filename changed to `text_accessibility.js`, to more clearly indicate what it does?

In any case, given how this is now initialized and used, I think this file should be moved into the `web/`-folder instead.